### PR TITLE
Feat: interceptor for invite service

### PR DESCRIPTION
**Pull request by skelet0no**
1. Added custom interceptor
2. Connected the interceptor to invite service
3. Fixed status code in invite service impl

### CODE REVIEW

File: invite_service/src/invite_service_impl.py [hunk 1]
```diff
@@ -366,25 +294,18 @@ async def update_invite(
...
-            await self._invite_repository.update_invite(invite=invite)
-            context.set_code(grpc.StatusCode.OK)
-            return proto.BaseResponse(status_code=200)
-        except ValueNotFoundError:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-            return proto.BaseResponse(status_code=404, message="Invite not found")
-        except prisma.errors.PrismaError:
-            context.set_code(grpc.StatusCode.INTERNAL)
-            return proto.BaseResponse(status_code=500, message="Internal server error")
+        invite = Invite.from_grpc_invite(request.invite)
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:306]: Add permission check

File: invite_service/src/invite_service_impl.py [hunk 2]
```diff
@@ -479,26 +385,21 @@ async def delete_invites_by_author_id(
...
-            await self._invite_repository.delete_invites_by_author_id(
-                author_id=request.author_id
-            )
-            return proto.BaseResponse(status_code=200)
-        except ValueNotFoundError:
-            return proto.BaseResponse(status_code=404, message="No invites found")
-        except prisma.errors.PrismaError:
-            context.set_code(grpc.StatusCode.INTERNAL)
-            return proto.BaseResponse(status_code=500, message="Internal server error")
+        await self._invite_repository.delete_invites_by_author_id(
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:413]: Add permission check

File: invite_service/src/invite_service_impl.py [hunk 3]
```diff
@@ -511,17 +412,12 @@ async def delete_invites_by_invitee_id(
...
-            await self._invite_repository.delete_invites_by_invitee_id(
-                invitee_id=request.invitee_id
-            )
-            return proto.BaseResponse(status_code=200)
-        except ValueNotFoundError:
-            return proto.BaseResponse(status_code=404, message="No invites found")
-        except prisma.errors.PrismaError:
-            context.set_code(grpc.StatusCode.INTERNAL)
-            return proto.BaseResponse(status_code=500, message="Internal server error")
+        await self._invite_repository.delete_invites_by_invitee_id(
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:445]: Add permission check

File: invite_service/utils/custom_interceptor.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# mypy: ignore-errors
+"""Interceptor decorator"""
```
Comment (Nhsdkk) [inline invite_service/utils/custom_interceptor.py]: Its not a decorator but rather just interceptor

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Good job, but its not enough. Make some changes

File: invite_service/src/invite_service_impl.py [hunk 5]
```diff
@@ -74,23 +71,15 @@ async def get_invites_by_event_id(
...
-                invites=proto.ListOfInvites(invites=[invite.to_grpc_invite() for invite in invites]))
-        except ValueNotFoundError:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-            return proto.InvitesResponse(status_code=404, message="Invites not found")
-        except prisma.errors.PrismaError:
-            context.set_code(grpc.StatusCode.INTERNAL)
-            return proto.InvitesResponse(
-                status_code=500, message="Internal server error"
-            )
+        invites = await self._invite_repository.get_invites_by_event_id(
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:74]: add pagination

File: invite_service/src/invite_service_impl.py [hunk 6]
```diff
@@ -366,25 +294,23 @@ async def update_invite(
...
-            return proto.BaseResponse(status_code=200)
-        except ValueNotFoundError:
-            context.set_code(grpc.StatusCode.NOT_FOUND)
-            return proto.BaseResponse(status_code=404, message="Invite not found")
-        except prisma.errors.PrismaError:
-            context.set_code(grpc.StatusCode.INTERNAL)
-            return proto.BaseResponse(status_code=500, message="Internal server error")
+        invite = Invite.from_grpc_invite(request.invite)
+        if (
+            request.requesting_user.id != invite.author_id
```
Comment (Nhsdkk) [inline invite_service/src/invite_service_impl.py:308]: Invitee also can edit the invite

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Review by MaxPlays35 — APPROVED** [review]

**Review by ZixMai — APPROVED** [review]

**Pull request merged**